vyper/codegen/stmt.py

```diff
@@ -155,7 +155,10 @@ def _assert_reason(self, test_expr, msg):
             ir_node = revert_seq
         else:
             ir_node = ["if", ["iszero", test_expr], revert_seq]
-        return IRnode.from_list(ir_node, error_msg="user revert with reason")
+
+        ir_node = IRnode.from_list(ir_node)
+        ir_node.set_error_msg("user revert with reason")
```

Comment:
what if some of the arguments have their own error message? Then it gets overwritten although it might be more specific. 

eg in the case of `raise self.bar(5 % i)` the mod is annotated with a `safemod` error msg, which will be replaced with `user revert with reason`

Reply:
Hmm I did implement set_error_msg to override, but that could be changed. @sandbubbles could you change that in this PR to not override? And add @cyberthirst's test case `raise self.foo(5%i)` to the test suite.

Reply:
does it make sense actually to just call `set_error_msg` in `from_list()`? since i imagine the same issue may happen at other call sites?

Reply:
resolved as of https://github.com/vyperlang/vyper/pull/4286/commits/f999cf653f3b0867f2ccfd00773ffdbb7e78005a